openpype/hosts/flame/api/plugin.py

```diff
@@ -1,3 +1,618 @@
+                data[k]["value"] = self.create_row(
+                    content_layout, "QCheckBox", v["label"],
+                    setChecked=v["value"], setToolTip=tool_tip)
+            elif v["type"] == "QSpinBox":
+                data[k]["value"] = self.create_row(
+                    content_layout, "QSpinBox", v["label"],
+                    setValue=v["value"], setMinimum=0,
+                    setMaximum=100000, setToolTip=tool_tip)
+        return data
+
+
+class Spacer(QtWidgets.QWidget):
+    def __init__(self, height, *args, **kwargs):
+        super(self.__class__, self).__init__(*args, **kwargs)
+
+        self.setFixedHeight(height)
+
+        real_spacer = QtWidgets.QWidget(self)
+        real_spacer.setObjectName("Spacer")
+        real_spacer.setFixedHeight(height)
+
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.addWidget(real_spacer)
+
+        self.setLayout(layout)
+
+
+class Creator(openpype.Creator):
+    """Creator class wrapper
+    """
+    clip_color = lib.ctx.color_map["purple"]
+    rename_index = None
+
+    def __init__(self, *args, **kwargs):
+        super(Creator, self).__init__(*args, **kwargs)
+        self.presets = openpype.get_current_project_settings()[
+            "flame"]["create"].get(self.__class__.__name__, {})
+
+        # adding basic current context flame objects
+        self.project = lib.get_current_project()
+        self.sequence = lib.get_current_sequence(opflame.selection)
+
+        if (self.options or {}).get("useSelection"):
+            self.selected = lib.get_sequence_segments(self.sequence, True)
+        else:
+            self.selected = lib.get_sequence_segments(self.sequence)
+
+        self.widget = CreatorWidget
+
+
+class PublishableClip:
+    """
+    Convert a segment to publishable instance
+
+    Args:
+        segment (flame.PySegment): flame api object
+        kwargs (optional): additional data needed for rename=True (presets)
+
+    Returns:
+        flame.PySegment: flame api object
+    """
+    vertical_clip_match = {}
+    marker_data = {}
+    types = {
+        "shot": "shot",
+        "folder": "folder",
+        "episode": "episode",
+        "sequence": "sequence",
+        "track": "sequence",
+    }
+
+    # parents search patern
+    parents_search_patern = r"\{([a-z]*?)\}"
+
+    # default templates for non-ui use
+    rename_default = False
+    hierarchy_default = "{_folder_}/{_sequence_}/{_track_}"
+    clip_name_default = "shot_{_trackIndex_:0>3}_{_clipIndex_:0>4}"
+    subset_name_default = "[ track name ]"
+    review_track_default = "[ none ]"
+    subset_family_default = "plate"
+    count_from_default = 10
+    count_steps_default = 10
+    vertical_sync_default = False
+    driving_layer_default = ""
+
+    def __init__(self, cls, segment, **kwargs):
+        # populate input cls attribute onto self.[attr]
+        self.__dict__.update(cls.__dict__)
```

Comment:
The `cls` is actualy not class but object of creator so `creator` would make more sense.

I would say this is super confusing like star imports and dangerous as hell hoping that it will not accedentally change something that should not be. If you need to use something from the creator object you should use it explicitly and not pass all attributes of creator to object of this class (which will change all attributes and methods). If it's on purpose then it's impossible to understand what and why is here happening...

Reply:
yeah, I know, but I need to udpate some creator attributes from within this class so they will be dynamic. This is mainly due shot rename and I need to hold actual shot index for renamer.

Reply:
so I guess this way it is beter.. gitkraken://repolink/33c731ef76967f4bd3f6ba8b287ecc0c52412f33/commit/ae02ad0d86fc7cf1207017729db24e7b48f326ad?url=https%3A%2F%2Fgithub.com%2Fpypeclub%2FOpenPype.git